src/main/java/org/zalando/nakadi/service/EventStream.java

```diff
@@ -142,30 +146,47 @@ public void streamEvents(final AtomicBoolean connectionReady) {
                 .collect(Collectors.toMap(identity(), valueFunction));
     }
 
-    public static String createStreamEvent(final Cursor cursor, final List<String> events) {
-        final StringBuilder builder = new StringBuilder()
-                .append("{\"cursor\":{\"partition\":\"").append(cursor.getPartition())
-                .append("\",\"offset\":\"").append(cursor.getOffset()).append("\"}");
+    public static int writeStreamEvent(final OutputStream outputStream, final Cursor cursor, final List<byte[]> events)
+            throws IOException {
+        final CountingOutputStream countingOutputStream = new CountingOutputStream(outputStream) {
+            @Override
+            public void flush() throws IOException {
+                // block flushing
```

Comment:
Since this is used as a wrapper, it will just block the flush call from the wrapped OutputStream to propagate to the "upstream". It would be cleaner code to have a separate wrapper for blocking the flushes. 

The reason to do this is that when the OutputStreamWriter is flushed so that that flush call doesn't propagate and cause unwanted side effects. OutputStreamWriter has to be flushed after it's use is over since it will internally buffer bytes. 

I guess that extracting more classes and adding more unit tests would be necessary to make the code and logic understandable.